galley/pkg/config/event/transformer.go

```diff
@@ -38,3 +41,93 @@ type Transformer interface {
 	// Outputs for this transformer
 	Outputs() collection.Names
 }
+
+// FnTransform is a base type for handling common Transformer operations.
+type FnTransform struct {
+	in       collection.Names
+	out      collection.Names
+	selector Selector
+	startFn  func()
+	stopFn   func()
+	handleFn func(e Event, h Handler)
+	syncCtr  int32
+}
+
+// Inputs partially implements Transformer
+func (t *FnTransform) Inputs() collection.Names {
+	return t.in
+}
+
+// Outputs partially implements Transformer
+func (t *FnTransform) Outputs() collection.Names {
+	return t.out
+}
+
+// Start implements Transformer
+func (t *FnTransform) Start() {
+	scope.Processing.Debug("FnTransform.Start")
+	if t.selector == nil {
+		t.selector = NewSelector()
+	}
+
+	atomic.StoreInt32(&t.syncCtr, int32(len(t.in)))
+
+	if t.startFn != nil {
+		t.startFn()
+	}
+}
+
+// Stop implements Transformer
+func (t *FnTransform) Stop() {
+	scope.Processing.Debug("FnTransform.Stop")
+	if t.stopFn != nil {
+		t.stopFn()
+	}
+}
+
+// Select partially implements Transformer
+func (t *FnTransform) Select(c collection.Name, h Handler) {
```

Comment:
`Select` is a confusing name for this ... maybe just use `Register` or `AddRoute`?